src/lib.rs

```diff
@@ -40,8 +43,9 @@ impl Xs {
         let num_ptr: *mut c_uint = &mut num;
```

Comment:
nit: It's not directly part of the PR but imho it is confusing nonetheless. I would drop the `num` var and only use the pointer. Also, what is `num_ptr` supposed to be? Isn't this just the array length?

Reply:
fixed by 818d62e